tools/ci-ts/__mocks__/dockerode.ts

```diff
@@ -30,14 +14,13 @@ const container = {
     .mockImplementation(async () => ({ State: containerState })),
 }
 
-const docker = {
-  getContainer: jest.fn().mockReturnValue(container),
-}
+const getContainer = jest.fn().mockReturnValue(container)
 
-const generator = () => jest.fn().mockReturnValue(docker)
-const mock = (jest.mock('dockerode', generator) as unknown) as DockerodeMock
+class DockerodeMock {
+  getContainer = getContainer
+  static setContainerState = setContainerState
```

Comment:
Lets inline the static method rather than assigning them to functions we defined earlier

Reply:
@HenryNguyen5 they need to be the same function object so that we can figure out whether it's been called or not in the test. Basically, the instance method actually mocks the functionality of dockerode, and the static methods are "cheat" methods on the mocks that help us in the tests.

Reply:
ahh okay I can do it like this...
```ts
  static getContainer = jest.fn().mockReturnValue(container)
  getContainer = DockerodeMock.getContainer

```